q2cli/core/usage.py

```diff
@@ -0,0 +1,73 @@
+import qiime2.sdk.usage as usage
+
+
+class CLIUsage(usage.Usage):
+    def __init__(self):
+        super().__init__()
+        self._imports = set()
+        self._recorder = []
+        self._init_data_refs = dict()
+
+    def _init_data_(self, ref, factory):
+        self._init_data_refs[ref] = factory
+        # Don't need to compute anything, so just pass along the ref
+        return ref
+
+    def _merge_metadata_(self, ref, records):
+        first_md = records[0].ref
+        remaining_records = ', '.join([r.ref for r in records[1:]])
+        t = '%s = %s.merge(%s)\n' % (ref, first_md, remaining_records)
```

Comment:
When you get around to implementing this, its worth pointing out that metadata merging in this interface is implicit - if metadata is provided multiple times, it is merged:

```bash
--m-metadata-file foo.tsv
--m-metadata-file bar.tsv
--m-metadata-file baz.tsv
```

This will merge all three metadata files.